### DISCUSSION TITLE
Right tool for the job?

### DISCUSSION THREAD

**Discussion by toasterparty** [nesrak1/AssetsTools.NET#96] [Q&A] [answered]
For my project I need to automate the following process:
- Open a `.bundle` file built with Unity v2019.4.5 (IL2CPP)
- Repackage the contents (prefabs) in a way that a Unity v2017.4.8 (Mono) application can understand
- [Load and instantiate](https://docs.unity3d.com/ScriptReference/AssetBundle.LoadFromFile.html) prefabs from the context of that Unity v2017.4.8 (Mono) application

Is this something that AssetTools can help with? Any pitfalls you foresee?

I'm new to unity, not new to modding.

**Comment by nesrak1** [comment, marked as answer]
The Mono/IL2Cpp part isn't really relevant, but the conversion from 2019 to 2017 will be.

If you pretend the assets are like json files, the format of those json files will be different between 2017 and 2019. In one version, you might have data in one array, whereas in another, it will be split into two separate arrays. The point is, you have to manually do the conversion between these two formats for every type of asset that exists. This isn't something that AssetsTools.NET (or any other tool I know of) can do (it's just a lot of work). The next best thing is putting it into AssetRipper, converting it back to a Unity project, and rebuilding it. Although downgrading a project/bundle with the Unity editor probably won't work very well either, especially with versions that far away from each other.
